src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -11,35 +11,50 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Task = System.Threading.Tasks.Task;
+using Microsoft.VisualStudio.Threading;
 using NuGet.PackageManagement.UI;
+using Microsoft.VisualStudio.Shell;
 
 namespace NuGet.PackageManagement.VisualStudio
 {
     public static class ProjectJsonToPackageRefMigrator
     {
         public static async Task MigrateAsync(LegacyCSProjPackageReferenceProject project, string dteProjectFullName)
         {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
> Which is why when I surely know i dont need UI thread, I better be with BG thread.

Yes, that's a fine policy. But from the last few PRs I've seen for this codebase, I almost always find a few violations of this where code should have been on the UI thread but wasn't. So I advise more caution going forward.